app/services/moab_storage_service.rb

```diff
@@ -2,6 +2,20 @@
 
 # Responsibility: interactions with Moab storage to support read-only access for ReST API calls
 class MoabStorageService
+  # @param [String] druid
+  # @param [String] category category of desired file ('content', 'metadata', or 'manifest')
+  # @param [String] filename name of the file (path relative to base directory)
+  # @param [String] version of the file (nil = latest)
+  # @return [Pathname] Pathname object containing the full path for the specified file
+  # @raise [Moab::ObjectNotFoundException] if file not found
+  def self.retrieve_file(druid, category, filename, version=nil)
+    raise(ArgumentError, "No filename provided to MoabStorageService.retrieve_file for druid #{druid}") if filename.blank?
+    err_msg = "category arg must be 'content', 'metadata', or 'manifest' (MoabStorageService.retrieve_file for druid #{druid})"
+    raise(ArgumentError, err_msg) unless ['content', 'metadata', 'manifest'].include?(category)
+    file_pathname = Stanford::StorageServices.retrieve_file(category, filename, druid, version)
+    File.open(file_pathname).read
```

Comment:
This would read a whole file into memory.  Is that going to be a problem?  Do you want to stream the file instead?

Reply:
could stream, sure.

Reply:
hrmm - pardon my ignorance but ... how do we stream a file to the API consumer?  Reading the file in chunks I get, but would we we serve it out in chunks to the API consumer, OR how do we create a single object to serve out without it being in memory?

Reply:
I would have this method return `File.open()` which is an IO object rather than `File.open().read` which is a String.

Reply:
great - thanks.  My ignorance has shrunk an iota.